api/src/pcapi/core/users/repository.py

```diff
@@ -110,10 +110,13 @@ def get_newly_eligible_age_18_users(since: date) -> list[models.User]:
             models.User.has_beneficiary_role == False,  # not already beneficiary
             models.User.has_admin_role == False,  # not an admin
             offerers_models.UserOfferer.userId.is_(None),  # not a pro
-            models.User.dateOfBirth > today - relativedelta(years=(constants.ELIGIBILITY_AGE_18 + 1)),  # less than 19yo
-            models.User.dateOfBirth <= today - relativedelta(years=constants.ELIGIBILITY_AGE_18),  # more than or 18yo
-            models.User.dateOfBirth
-            > since - relativedelta(years=constants.ELIGIBILITY_AGE_18),  # less than 18yo at since
+            # less than 19yo
+            models.User.birth_date > today - relativedelta(years=(constants.ELIGIBILITY_AGE_18 + 1)),  # type: ignore [operator]
+            # more than or 18yo
+            models.User.birth_date <= today - relativedelta(years=constants.ELIGIBILITY_AGE_18),  # type: ignore [operator]
+            # less than 18yo at since
+            models.User.birth_date  # type: ignore [operator]
+            > since - relativedelta(years=constants.ELIGIBILITY_AGE_18),
             models.User.dateCreated < today,
```

Comment:
Ce filtre était déjà là, mais ne correspond pas à la docstring, qui dit : 

> that have created their account before `since`

Je ne sais pas qui du code ou de la docstring a raison, je suppose que ça dépend de l'usage qu'on fait de cette fonction.

(La correction devrait avoir lieu dans une PR séparée, car le sujet est complètement distinct du sujet de cette PR.)